src/library/columncache.cpp

```diff
@@ -81,7 +81,7 @@ void ColumnCache::setColumns(const QStringList& columns) {
     m_columnSortByIndex.insert(m_columnIndexByEnum[COLUMN_LIBRARYTABLE_GENRE], sortNoCase);
     m_columnSortByIndex.insert(m_columnIndexByEnum[COLUMN_LIBRARYTABLE_COMPOSER], sortNoCase);
     m_columnSortByIndex.insert(m_columnIndexByEnum[COLUMN_LIBRARYTABLE_GROUPING], sortNoCase);
-    m_columnSortByIndex.insert(m_columnIndexByEnum[COLUMN_LIBRARYTABLE_TRACKNUMBER], sortInt);
+    m_columnSortByIndex.insert(m_columnIndexByEnum[COLUMN_LIBRARYTABLE_TRACKNUMBER], sortNoCase);
```

Comment:
Currently the string found in the tags is preserved as much as possible. The same is true if a user edits track numbers in Mixxx.

I would propose to add a hidden Integer column "tracknumber_current" where we store only the current track number as an Integer if it can be parsed successfully from the track number string. This column should only be used internally in database queries for sorting purposes and should not be visible to the user.

Later we might decide to completely drop the track number string and split it into 2 integer values "tracknumber_current" and "tracknumber_total".

Are the terms "current" and "total" reasonable? I didn't find a more appropriate alternative for "current".